### ISSUE TITLE
Unable to load HEIC file since commit 16c633d

### ISSUE THREAD

**Issue by julianfairfax** [flathub/org.gnome.eog#73]
Hi,

I am running this Flatpak on Arch, and I am unable to view HEIC images in Image Viewer starting from https://github.com/flathub/org.gnome.eog/commit/16c633d413d71c6e23368a0aed61ede4ed10a340. Downgrading fixes this, but is obviously not a solution.

Error log:
```
** (eog:2): WARNING **: 11:49:01.937: Cannot read plugin directory.

(eog:2): GdkPixbuf-CRITICAL **: 11:49:01.938: gdk_pixbuf_animation_is_static_image: assertion 'GDK_IS_PIXBUF_ANIMATION (animation)' failed

(eog:2): GdkPixbuf-CRITICAL **: 11:49:01.938: gdk_pixbuf_animation_get_iter: assertion 'GDK_IS_PIXBUF_ANIMATION (animation)' failed

(eog:2): GdkPixbuf-CRITICAL **: 11:49:01.938: gdk_pixbuf_animation_iter_get_pixb
```

**Comment by TingPing** [comment]
All we can do is downgrade it. Bugs should be reported upstream.

**Comment by TingPing** [comment]
Downgraded it for now.

**Comment by farindk** [comment]
libheif author here.
I don't know much about how the flatpak packaging works and don't use it myself, but it looks to me like a packaging issue.
Since v1.14.0 the codec backends of libheif can be built into external `.so` plugin files that are then loaded dynamically.
By default, these are expected in `LIBHEIF_PLUGIN_PATH`, or, if that is not set, in `/usr/local/lib/libheif/' or whatever was specified during the build.

Since the Flatpak says "Cannot read plugin directory", I presume that either the plugins are not installed, or the plugin path is set incorrectly.

As a quick workaround, one can disable the use of plugins in the build settings.

**Comment by TingPing** [comment]
@farindk Thanks for the hints, I'll look into it on our side :)

**Comment by TingPing** [comment]
@farindk Well I fixed one bug: https://github.com/strukturag/libheif/pull/743

However these are not actually installed as plugins currently, they are built-in modules.

**Comment by TingPing** [comment]
Fixed in https://github.com/strukturag/libheif/pull/744 - will land in this package with next upstream release.

**Comment by farindk** [comment]
@TingPing: since you are building all codecs built-in, another option would be to set the cmake variable `ENABLE_PLUGIN_LOADING` to off. Then, it's not required to have an empty plugin directory.

**Comment by TingPing** [comment]
Good point. I've updated this package with 1.14.0 again with plugins disabled, works fine.

Thanks for the reviews.
